packages/lit-dev-content/site/docs/frameworks/react.md

```diff
@@ -0,0 +1,153 @@
+---
+title: React
+eleventyNavigation:
+  key: React
+  parent: Framework integration
+  order: 1
+  labs: true
+---
+
+{% labs-disclaimer %}
+
+The [@lit-labs/react](https://github.com/lit/lit/tree/main/packages/labs/react) package provides web component and reactive controller integration for React. While React can render web components as-is, there are a few gaps in the developer experience. You can't easily pass React props to custom element properties or add event listeners to custom elements. (For more information on the limitations of React's web component integration, see [Custom Elements Everywhere](https://custom-elements-everywhere.com/libraries/react/results/results.html).)
+
+The `@lit-labs/react` package provides two main entry points:
+
+-   `createComponent()` creates a React component that _wraps_ an existing web component. The wrapper allows you to set props on the component and add event listeners to the component like you would any other React component.
+
+-   `useController()` lets you use a Lit reactive controller as a React hook.  
+
+## createComponent
+
+The `createComponent()` function makes a React component wrapper for a custom element class. The wrapper correctly passes React `props` to properties accepted by the custom element and listens for events dispatched by the custom element.
+
+### How it works
+
+For properties, the wrapper interrogates the web component class to discover its available properties. Then any React props passed with property names are set on the custom element as properties and not attributes.
```

Comment:
```suggestion
During a render, the wrapper receives props from React and based on the options and the custom element class, changes the behavior of some of the props:

* If a prop name is a property on the custom element, as determined with an `in` check, the wrapper sets that property on the element to the prop value
* If a prop name is an event name passed to the `events` option, the prop value is passed to `addEventListener()` with the name of the event.
* Otherwise the prop is passed to React's `createElement()` to be rendered as an attribute.

Both properties and events are added in `componentDidMount()` and `componentDidUpdate()` callbacks, because the element must have already been instantiated by React in order to access it.
```